File: avent_code1.py
# Solve Avent of Code Day 1
# Lauryne Feuvrier _ 25.09.25

from collections import Counter

# Partie 1: 

Col_G = []
Col_D = []

with open("input1.txt") as file:
    for line in file:
        g, d = line.split()
        Col_G.append(int(g))
        Col_D.append(int(d))

Col_G.sort()
Col_D.sort()

res1 = sum(abs(a - b) for a, b in zip(Col_G, Col_D))
print(res1)

# Partie 2:

counter_D = Counter(Col_D)

res2 = 0

for a in Col_G :
    res2 += a * counter_D.get(a,0)

print(res2)

File: avent-code1.cpp
#include <iostream>
#include <vector>
#include <string>
#include <fstream>
#include <algorithm>

using namespace std;

int main()
{
    int g,d;
    long long res1=0, res2=0;
    vector<int> ColG, ColD; 

    ifstream f("input1.txt"); // ouvre le fichier

    if (!f.is_open())
        cout << "Impossible d'ouvrir le fichier en lecture !" << endl;
    else
        while(f >> g >> d){
            ColG.push_back(g);
            ColD.push_back(d);
        }
    
    sort(ColG.begin(), ColG.end());
    sort(ColD.begin(), ColD.end());

    for(int i = 0; i < ColG.size();i++){
        res1+= abs(ColG[i]-ColD[i]);

        // Partie 2 
        int Counted_inD = count(ColD.begin(), ColD.end(), ColG[i]);
        res2+= ColG[i]*Counted_inD;
        
    }

    cout << res1 << endl;
    cout << res2 << endl;

    return 0;
}